Oxfmt Missing type exports cause TS4058 when re-exporting defineConfig

When a downstream library wraps and re-exports defineConfig (or any function returning OxfmtConfig), TypeScript's declaration emit fails with TS4058 because several interfaces used by OxfmtConfig are not exported from oxfmt.

The types OxfmtOverrideConfig, SortImportsConfig, SortPackageJsonConfig, and SortTailwindcssConfig are declared as interfaces in dist/index.d.ts and used as property types of the exported OxfmtConfig interface, but they are missing from the export statement.

## oxfmt version

`0.41.0`

## Reproduction

```ts
import { defineConfig, OxfmtConfig } from "oxfmt";
export function extendsOxfmtConfig(config: OxfmtConfig) {
  return defineConfig({
    arrowParens: "always",
    bracketSpacing: true,
    printWidth: 80,
    ...config,
  });
}
```
Build with any tool that generates `.d.ts` files (e.g. `tsdown`, `tsc --declaration`).

### Error

```
error TS4058: Return type of exported function has or is using name 'OxfmtOverrideConfig' from external module ".../node_modules/oxfmt/dist/index" but cannot be named.
error TS4058: Return type of exported function has or is using name 'SortImportsConfig' from external module ".../node_modules/oxfmt/dist/index" but cannot be named.
error TS4058: Return type of exported function has or is using name 'SortPackageJsonConfig' from external module ".../node_modules/oxfmt/dist/index" but cannot be named.
error TS4058: Return type of exported function has or is using name 'SortTailwindcssConfig' from external module ".../node_modules/oxfmt/dist/index" but cannot be named.
```

## Root cause

In `dist/index.d.ts`, the export statement on line 676 is:

```ts
export { type FormatConfig, FormatOptions, OxfmtConfig, SortImportsOptions, SortPackageJsonOptions, SortTailwindcssOptions, TailwindcssOptions, defineConfig, format, jsTextToDoc };
```

The following interfaces are used by `OxfmtConfig` but not included in the export:

- `OxfmtOverrideConfig` (used by `OxfmtConfig.overrides`)
- `SortImportsConfig` (used by `OxfmtConfig.sortImports`)
- `SortPackageJsonConfig` (used by `OxfmtConfig.sortPackageJson` / `SortPackageJsonUserConfig`)
- `SortTailwindcssConfig` (used by `OxfmtConfig.sortTailwindcss`)

## Expected behavior

All types transitively reachable from exported types/functions should also be exported. The export statement should include:

```ts
export {
  type FormatConfig,
  FormatOptions,
  OxfmtConfig,
  OxfmtOverrideConfig,        // missing
  SortImportsConfig,           // missing
  SortImportsOptions,
  SortPackageJsonConfig,       // missing
  SortPackageJsonOptions,
  SortTailwindcssConfig,       // missing
  SortTailwindcssOptions,
  TailwindcssOptions,
  defineConfig,
  format,
  jsTextToDoc,
};
```

## Workaround

Downstream consumers can explicitly annotate the return type to avoid declaration emit needing to reference the unexported types:

```ts
import { defineConfig, OxfmtConfig } from "oxfmt";
export function extendsOxfmtConfig(config: OxfmtConfig): OxfmtConfig {
  return defineConfig({
    arrowParens: "always",
    ...config,
  });
}
```

## Environment
- oxfmt: 0.41.0
- TypeScript: 5.9.3
- tsdown: 0.21.4 (uses rolldown-plugin-dts 0.22.5)
- Node: >= 22.12.0

--- a/apps/oxfmt/src-js/index.ts
+++ b/apps/oxfmt/src-js/index.ts
@@ -18,11 +18,15 @@ import type {
 } from "./config.generated";
 
 // --- Type exports ---
-// Only exports top level types for now
 
-// The same naming convention as `oxlint` for consistency
-export type OxfmtConfig = Oxfmtrc;
-export type { FormatConfig } from "./config.generated";
+// Re-export all generated config types.
+// So that downstream libraries can reference them in declaration emit without TS4058/TS4082 errors.
+export type * from "./config.generated";
+
+// The same naming convention as `oxlint` for consistency.
+// Using `interface extends` so that TypeScript displays `OxfmtConfig` in errors
+// and hovers instead of resolving to the generated `Oxfmtrc` name.
+export interface OxfmtConfig extends Oxfmtrc {}
 
 // Backward-compatible type aliases using `Options` suffix.
 

